mp/src/game/server/neo/neo_player.cpp

```diff
@@ -2288,7 +2289,18 @@ bool CNEO_Player::ProcessTeamSwitchRequest(int iTeam)
 			}
 		}
 
-		StopObserverMode();
+		if(IsDead())
+		{
```

Comment:
Moving some of the logic to a think, similar to what the clientside `c_neo_player.cpp` does with `m_bFirstAliveTick` might be an option for getting around this

Reply:
Thanks, I am taking a look 👀

Reply:
> This condition is passed before the client materializes in the world, making it impossible to spawn in the level in some situations.
This was because players could not spawn once their state was set to Observer. Relying on this being true probably caused the issue where you could respawn the first time you died any time during a live round on a new map.
This probably also fixes the TDM spawn issue.